### PR TITLE
LPS-176829 | Test Automation | Master

### PR DESCRIPTION
**Pull request by pedroamarall**
Ticket:
https://issues.liferay.com/browse/LPS-176829

### CODE REVIEW

**Comment by pedroamarall** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 1]
```diff
@@ -2806,6 +2806,38 @@ definition {
 		}
 	}
 
+	@description = "LPS-176829 - Verify that Formula fields can't be required"
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:2809]: On @decription, use LPS from User Story: `LPS-164948`

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 2]
```diff
@@ -2806,6 +2806,38 @@ definition {
...
+	@priority = 4
+	test FormulaFieldsIsNotMandatory {
+		property portal.acceptance = "true";
+
+		task ("Given: an Object") {
+			ObjectAdmin.addObjectViaAPI(
+				labelName = "Custom Object test",
+				objectName = "CustomObjecttest",
+				pluralLabelName = "Custom Objects test");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:2818]: In the `addObjectviaAPI`, we use the `Custom Object + LPS from the automation ticket`, this way we avoid errors in the CI due to duplicate objects.

Ex:
```
ObjectAdmin.addObjectViaAPI(
	labelName = "Custom Object 176829",
	objectName = "CustomObject176829",
	pluralLabelName = "Custom Objects 176829");
```

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 3]
```diff
@@ -2806,6 +2806,38 @@ definition {
...
+
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectSystemObject(label = "Custom Object test");
```
Comment (andreefarias) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:2822]: Change here too.

**Review by andreefarias — CHANGES_REQUESTED** [review]
Good job, @pedroamarall!

I requested only some changes. Fix it, please. Thanks!

**Comment by pedroamarall** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 4]
```diff
@@ -2806,6 +2806,38 @@ definition {
...
+		}
+
+		task ("When: adding a Formula Field") {
+			ObjectAdmin.addObjectFieldViaUI(
+				fieldLabel = "Custom Formula Test",
+				fieldType = "Formula",
+				formulaFieldOption = "Decimal");
+		}
+
+		task ("Then:It's not possible to select mandatory option") {
```
[the remark this answers is not in the document]
Comment (pedroamarall) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase]: Done @andreefarias, Thanks for your feedback

**Comment by andreefarias** [comment]
Good job! Thanks for the collaboration @pedroamarall! 🎉 

I made some changes because it is necessary to assert the input as well, but I didn't write it in the steps, so I corrected it myself. Thanks!

**Comment by andreefarias** [comment]
New PR [here](https://github.com/liferay-bpm-qa/liferay-portal/pull/617)

**Comment by andreefarias** [comment]
ci:close

**Pull request closed without merging**